From 180ddb608d30214a71dc16d558fbb43daa32c47b Mon Sep 17 00:00:00 2001
From: Gabriele M <moto.falcon.git@gmail.com>
Date: Mon, 14 Nov 2016 22:15:38 +0100
Subject: [PATCH 1/5] GsmCdmaPhone: Return dummy ICCID serial for NV sub

In case of NV sub there's no ICC, so both getIccSerialNumber()
and getFullIccSerialNumber() return null.

Since getFullIccSerialNumber() is used to set/get the ID of
each PhoneAccountHandle, devices with NV subs can't register
proper phone accounts other than the emergency one.

getIccSerialNumber() is instead used by CarrierConfigLoader
to store and load configs.

Fix both the problems forcing dummy serials for NV subscriptions.

Change-Id: I610115e9d9f5b3b86b36fddcf13dbed6387e377c
(cherry picked from commit 94ae0fe784f0701d3c3bb3c7575362dc6684300b)
---
 src/java/com/android/internal/telephony/GsmCdmaPhone.java | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/src/java/com/android/internal/telephony/GsmCdmaPhone.java b/src/java/com/android/internal/telephony/GsmCdmaPhone.java
index 972859cf6..5abcefb80 100644
--- a/src/java/com/android/internal/telephony/GsmCdmaPhone.java
+++ b/src/java/com/android/internal/telephony/GsmCdmaPhone.java
@@ -144,6 +144,7 @@ public class GsmCdmaPhone extends Phone {
     };
     public static final String PROPERTY_CDMA_HOME_OPERATOR_NUMERIC =
             "ro.cdma.home.operator.numeric";
+    private static final String DUMMY_NV_ICC_SERIAL = "DUMMY_NV_ICC_SERIAL";
 
     //CDMALTE
     /** PHONE_TYPE_CDMA_LTE in addition to RuimRecords needs access to SIMRecords and
@@ -728,6 +729,9 @@ public class GsmCdmaPhone extends Phone {
 
     @Override
     public String getIccSerialNumber() {
+        if (mCdmaSubscriptionSource == CDMA_SUBSCRIPTION_NV) {
+            return DUMMY_NV_ICC_SERIAL;
+        }
         IccRecords r = mIccRecords.get();
         if (!isPhoneTypeGsm() && r == null) {
             // to get ICCID form SIMRecords because it is on MF.
@@ -738,6 +742,9 @@ public class GsmCdmaPhone extends Phone {
 
     @Override
     public String getFullIccSerialNumber() {
+        if (mCdmaSubscriptionSource == CDMA_SUBSCRIPTION_NV) {
+            return DUMMY_NV_ICC_SERIAL;
+        }
         IccRecords r = mIccRecords.get();
         if (!isPhoneTypeGsm() && r == null) {
             // to get ICCID form SIMRecords because it is on MF.
-- 
2.17.0

